src/geocel/SurfaceParams.cc

```diff
@@ -42,24 +41,34 @@ SurfaceParams::SurfaceParams(inp::Surfaces const& input,
         // Convert the temporary data to device-compatible format
         HostVal<SurfaceParamsData> host_data;
         host_data.num_surfaces = labels_.size();
-        detail::VolumeSurfaceRecordBuilder build_record(
-            &host_data.volume_surfaces,
-            &host_data.volume_instance_ids,
-            &host_data.surface_ids);
-        std::for_each(temp_volume_surfaces.begin(),
-                      temp_volume_surfaces.end(),
-                      build_record);
+        if (host_data.num_surfaces > 0)
+        {
+            // Only resize surface array if surfaces are defined
+            detail::VolumeSurfaceRecordBuilder build_record(
+                &host_data.volume_surfaces,
+                &host_data.volume_instance_ids,
+                &host_data.surface_ids);
+            std::for_each(temp_volume_surfaces.begin(),
+                          temp_volume_surfaces.end(),
+                          build_record);
+        }
```

Comment:
```suggestion
        detail::VolumeSurfaceRecordBuilder build_record(
            &host_data.volume_surfaces,
            &host_data.volume_instance_ids,
            &host_data.surface_ids);
        std::for_each(temp_volume_surfaces.begin(),
                      temp_volume_surfaces.end(),
                      build_record);
```
We settled on creating empty boundary surface IDs if optical physics was enabled but no surfaces were defined, right? Otherwise we can't construct the `VolumeSurfaceView`.

Reply:
I've forgotten and confused myself: I guess "no surfaces" is the special case for optical physics, but the regular case for EM physics (which now will always also create surface params).

With the `if`, we can't instantiate a `VolumeSurfaceView`, which I thought was what we agreed but maybe not. Without the if, we have an unused volumes-length collection of volume-to-surface data. If we don't have surfaces, we won't have surface physics, and we shouldn't have to have to check the surface view at the boundary. But I suppose for optical physics, that would mean adding a special no-surface case.

So maybe I should restore the default constructor for surface physics—to be used if optical tracking is off and doesn't create any data—and implement the above change?

Reply:
> So maybe I should restore the default constructor for surface physics—to be used if optical tracking is off and doesn't create any data—and implement the above change?

I think that makes sense: empty surface data and no instantiating a `VolumeSurfaceView` if optical physics is off, and invalid boundary surfaces for any volumes that don't have surface physics if it's on (which could in the special case be all of them).